ddtrace/internal/datadog/profiling/dd_wrapper/src/profiler_stats.cpp

```diff
@@ -0,0 +1,67 @@
+#include "profiler_stats.hpp"
+
+#include <charconv>
+#include <string>
+
+namespace {
+
+void
+append_to_string(std::string& s, size_t value)
+{
+    char buf[128];
+    auto [ptr, ec] = std::to_chars(std::begin(buf), std::end(buf), value);
+    s.append(buf, ptr);
+}
+
+} // namespace
+
+void
+Datadog::ProfilerStats::increment_sampling_event_count(size_t k_sampling_event_count)
+{
+    sampling_event_count += k_sampling_event_count;
+}
+
+size_t
+Datadog::ProfilerStats::get_sampling_event_count()
+{
+    return sampling_event_count;
+}
+
+void
+Datadog::ProfilerStats::increment_sample_count(size_t k_sample_count)
+{
+    sample_count += k_sample_count;
+}
+
+size_t
+Datadog::ProfilerStats::get_sample_count()
+{
+    return sample_count;
+}
+
+void
+Datadog::ProfilerStats::reset_state()
+{
+    sample_count = 0;
+    sampling_event_count = 0;
+}
+
+std::string_view
+Datadog::ProfilerStats::get_internal_metadata_json()
+{
+    static std::string result;
```

Comment:
:thought_balloon: Implication is that you can never upload several profiles at the same time.
:bulb:  could it be just as easy to have a string buffer within the ProfilerStats class ? This will be tied to the profile and make it safe to manipulate several profiles at the same time.